lms/djangoapps/instructor/views/api.py

```diff
@@ -2700,51 +2703,45 @@ def send_email(request, course_id):
+            schedule = _convert_schedule_to_utc_from_local(schedule, browser_timezone, request.user)
+            _determine_valid_schedule(schedule)
+        except ValueError as error:
+            error_message = f"Error occurred while attempting to create a scheduled bulk email task: {error}"
+            log.error(f"{error_message}")
+            return HttpResponseBadRequest(repr(error_message))
 
-    # allow two branding points to come from Site Configuration: which CourseEmailTemplate should be used
-    # and what the 'from' field in the email should be
-    #
-    # If these are None (there is no site configuration enabled for the current site) than
-    # the system will use normal system defaults
-    course_overview = CourseOverview.get_from_id(course_id)
-    from_addr = configuration_helpers.get_value('course_email_from_addr')
-    if isinstance(from_addr, dict):
-        # If course_email_from_addr is a dict, we are customizing
-        # the email template for each organization that has courses
-        # on the site. The dict maps from addresses by org allowing
-        # us to find the correct from address to use here.
-        from_addr = from_addr.get(course_overview.display_org_with_default)
-
-    template_name = configuration_helpers.get_value('course_email_template_name')
-    if isinstance(template_name, dict):
-        # If course_email_template_name is a dict, we are customizing
-        # the email template for each organization that has courses
-        # on the site. The dict maps template names by org allowing
-        # us to find the correct template to use here.
-        template_name = template_name.get(course_overview.display_org_with_default)
+    # Retrieve the customized email "from address" and email template from site configuration for the course/partner. If
+    # there is no site configuration enabled for the current site then we use system defaults for both.
+    from_addr = _get_branded_email_from_address(course_overview)
+    template_name = _get_branded_email_template(course_overview)
 
-    # Create the CourseEmail object.  This is saved immediately, so that
-    # any transaction that has been pending up to this point will also be
-    # committed.
-    # TODO: convert to use bulk_email app's `create_course_email` API function and remove direct import and use of
-    # bulk_email model
+    # Create the CourseEmail object. This is saved immediately so that any transaction that has been pending up to this
+    # point will also be committed.
     try:
-        email = CourseEmail.create(
+        email = create_course_email(
             course_id,
             request.user,
             targets,
             subject,
             message,
             template_name=template_name,
-            from_addr=from_addr
+            from_addr=from_addr,
         )
     except ValueError as err:
-        log.exception('Cannot create course email for course %s requested by user %s for targets %s',
```

Comment:
This log message will be triggered in the `create_course_email` function now. This new function will re-raise the `ValueError` exception so it can be caught here and we can return the bad request response.